processing/src/main/java/org/apache/druid/query/topn/TopNQueryEngine.java

```diff
@@ -132,7 +132,7 @@ private TopNMapFn getMapFn(
       topNAlgorithm = new TimeExtractionTopNAlgorithm(adapter, query);
     } else if (selector.isHasExtractionFn()) {
       topNAlgorithm = new HeapBasedTopNAlgorithm(adapter, query);
-    } else if (columnCapabilities != null && !(columnCapabilities.getType() == ValueType.STRING
+    } else if (columnCapabilities == null || !(columnCapabilities.getType() == ValueType.STRING
```

Comment:
nit: maybe nice to comment when `columnCapabilities` can be null?

Reply:
added additional information to existing comment